src/wasm-lib/kcl/src/std/assert.rs

```diff
@@ -0,0 +1,121 @@
+//! Standard library assert functions.
+
+use anyhow::Result;
+use derive_docs::stdlib;
+use schemars::JsonSchema;
+
+use crate::{
+    errors::{KclError, KclErrorDetails},
+    executor::MemoryItem,
+    std::Args,
+};
+
+async fn _assert(value: bool, message: &str, args: &Args) -> Result<(), KclError> {
+    if !value {
+        return Err(KclError::Type(KclErrorDetails {
+            message: format!("assert failed: {}", message),
+            source_ranges: vec![args.source_range],
+        }));
+    }
+    Ok(())
+}
+
+/// Check that the provided value is true, or raise a [KclError]
+/// with the provided description.
+pub async fn assert(args: Args) -> Result<MemoryItem, KclError> {
+    let (data, description): (bool, String) = args.get_data()?;
+    args.make_user_val_from_f64(inner_assert(data, &description, args.clone()).await?)
```

Comment:
Can we do it without cloning `Args`? Can `inner_assert()` et al take `&Args`?

Reply:
@jtran very likely! I was copying the style from the other stdlib functions, but this was bugging me too. Maybe I'll just do it with borrows in assert and we can either cleanup the rest of the stdlib or someone can fix that back to owned if it was intentional for some reason